metadata-ingestion-modules/dagster-plugin/src/datahub_dagster_plugin/sensors/datahub_sensors.py

```diff
@@ -0,0 +1,462 @@
+from datahub.api.entities.dataset.dataset import Dataset
+from datahub.emitter.mce_builder import (
+    make_data_platform_urn,
+    make_dataplatform_instance_urn,
+)
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.emitter.rest_emitter import DatahubRestEmitter
+from datahub.metadata.schema_classes import DataPlatformInstanceClass, SubTypesClass
+from datahub.utilities.urns.dataset_urn import DatasetUrn
+
+from datahub_dagster_plugin.client.dagster_generator import (
+    DagsterEnvironment,
+    DagsterGenerator,
+    DagsterSourceConfig,
+)
+
+
+def make_datahub_sensor(
+    config: DagsterSourceConfig,
+    name: Optional[str] = None,
+    minimum_interval_seconds: Optional[int] = None,
+    description: Optional[str] = None,
+    job: Optional[ExecutableDefinition] = None,
+    jobs: Optional[Sequence[ExecutableDefinition]] = None,
+    default_status: DefaultSensorStatus = DefaultSensorStatus.STOPPED,
+    asset_selection: Optional[CoercibleToAssetSelection] = None,
+    required_resource_keys: Optional[Set[str]] = None,
+) -> SensorDefinition:
+    """Create a sensor on job status change emit lineage to DataHub.
+
+    Args:
+        config (DagsterSourceConfig): DataHub Sensor config
+        name: (Optional[str]): The name of the sensor. Defaults to "datahub_sensor".
+        minimum_interval_seconds: (Optional[int]): The minimum number of seconds that will elapse
+            between sensor evaluations.
+        default_status (DefaultSensorStatus): Whether the sensor starts as running or not. The default
+            status can be overridden from Dagit or via the GraphQL API.
+
+    Examples:
+        .. code-block:: python
+
+            datahub_sensor = make_datahub_sensor(
+               config
+            )
+
+            @repository
+            def my_repo():
+                return [my_job + datahub_sensor]
+
+
+    """
+
+    @sensor(
+        name=name,
+        minimum_interval_seconds=minimum_interval_seconds,
+        description=description,
+        job=job,
+        jobs=jobs,
+        default_status=default_status,
+        asset_selection=asset_selection,
+        required_resource_keys=required_resource_keys,
+    )
+    def datahub_sensor(context):
+        """
+        Sensor which instigate all run status sensors and trigger them based upon run status
+        """
+        for each in DatahubSensors(config).sensors:
+            each.evaluate_tick(context)
+        return SkipReason("Trigger run status sensors if any new runs present...")
+
+    return datahub_sensor
+
+
+class DatahubSensors:
+    def __init__(self, config: Optional[DagsterSourceConfig] = None):
+        """
+        Set dagster source configurations and initialize datahub emitter and dagster run status sensors
+        """
+        if config:
+            self.config = config
+        else:
+            self.config = DagsterSourceConfig()
+
+        self.emitter: DatahubRestEmitter = DatahubRestEmitter(
```

Comment:
let's create a graph instance here

that way we can also reuse the pydantic config models and not need to hardcode this entire list